Add configurable ordering for template groups

wachterjohannes opened the pull request:
| Q | A
| --- | ---
| Bug fix? | yes
| New feature? | no
| BC breaks? | yes (But with BC-Layer
| Deprecations? | yes (Additional arguments for `getGroups`)
| Fixed tickets | fixes none
| Related issues/PRs | none
| License | MIT
| Documentation PR | none

#### What's in this PR?

This change introduces the ability to configure custom ordering and translation keys for template groups via sulu_admin.yaml configuration.

#### Why?

This allows to reorder the template groups in the admin via configuration ad a

#### Example Usage

Configuration example:
```yaml
sulu_admin:
    template_groups:
        articles:
            blog:
                order: 1
                translation_key: 'app.blog_articles'
            news: 'app.news_articles'  # shorthand, order = array index
```

#### To Do

- [x] Add breaking changes to UPGRADE.md

wachterjohannes committed:
Add configurable ordering for template groups

This change introduces the ability to configure custom ordering and
translation keys for template groups via sulu_admin.yaml configuration.

Changes:
- Add 'order' property to FormGroup with default value of 9999
- Add 'withOrder()' method to FormGroup for immutability
- Add 'template_groups' configuration node with support for:
  - Full format: order + translation_key
  - Shorthand format: translation_key only (order defaults to array index)
- Update GroupProvider::getGroups() to accept optional resourceKey and
  templateType parameters (using variadic arguments for BC)
- Add deprecation warning when getGroups() is called without arguments
  (will be removed in Sulu 4.0)
- Groups without configuration use order 9999 and appear alphabetically
- Update UPGRADE-3.x.md with configuration examples and migration guide

Configuration example:
sulu_admin:
    template_groups:
        articles:
            blog:
                order: 1
                translation_key: 'app.blog_articles'
            news: 'app.news_articles'  # shorthand, order = array index

alexander-schranz commented:
Lets discuss this, we are more thinking of making this not configurable and go with Alphabetically as thats the only understandable way for all Content Manager.

wachterjohannes committed:
Add missing template_groups parameter to container

The template_groups configuration was defined in Configuration.php but
the parameter was not being set in the container, causing tests to fail
with "You have requested a non-existent parameter sulu_admin.template_groups".

This parameter is required by the metadata_group_provider service.

hual7 commented:
> Lets discuss this, we are more thinking of making this not configurable and go with Alphabetically as thats the only understandable way for all Content Manager.

Usually yes, but it would be nice to have a sorting option – e.g. in a multisite setup, where the admin of all webspaces have a list of sorted types. Furthermore, there may be inconsistencies in the admin translations as well if there is no configurable order.

And based on this order, there will be a more consistent order in "filter by types" in the smart content type.
<img width="672" height="594" alt="Bildschirmfoto 2025-12-05 um 12 24 42" src="https://github.com/user-attachments/assets/e1c238f5-0394-46c5-8428-f13f33dcf064" />

And sometimes the slogan "better to have than need" also applies  ;-)
